src/utils/retry.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/example-shopping
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {HttpErrors} from '@loopback/rest';
+import {promisify} from 'util';
+import * as debugFactory from 'debug';
+const debug = debugFactory('loopback:example:shopping');
+
+/**
+ * A task that can be retried
+ */
+export interface Task<T> {
+  run(): Promise<T | null | undefined>;
+  // tslint:disable-next-line:no-any
+  isDone?(err: any, result: T | null | undefined): result is T;
```

Comment:
[KISS](https://en.wikipedia.org/wiki/KISS_principle) & [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it).

Let's remove `isDone` from this pull request since we don't need it (yet).